### PR TITLE
Add hexString type

### PR DESCRIPTION
**Pull request by chmanie**
> ⚠️  NOTE: Please don't use the PR description for communication, use comments instead.

This PR adds a HexString type to handle the generated secret hash. It also
adjusts the documentation to a 1-3 point scale rather than (1-5).

### CODE REVIEW

File: docs/_API_ColonyClient.md [hunk 1]
```diff
@@ -57,15 +57,15 @@ Helper function used to generate the rating secret used in task ratings. Accepts
...
+|value|number|Value to hide (typically a rating of 1-3).|
 
 **Returns**
 
 A promise which resolves to an object containing the following properties:
 
 |Return value|Type|Description|
 |---|---|---|
-|secret|string|keccak256 hash of joint Salt and Value.|
+|secret|undefined|keccak256 hash of joint Salt and Value.|
```
Comment (JamesLefrere) [inline docs/_API_ColonyClient.md]: Looks like a missing type here

File: docs/_API_ColonyClient.md [hunk 2]
```diff
@@ -248,7 +248,7 @@ A promise which resolves to an object containing the following properties:
 
 |Return value|Type|Description|
 |---|---|---|
-|secret|string|the hashed rating (equivalent to the output of `keccak256(_salt, _rating)`).|
+|secret|undefined|the hashed rating (equivalent to the output of `keccak256(_salt, _rating)`).|
```
Comment (JamesLefrere) [inline docs/_API_ColonyClient.md]: And here

File: docs/_API_ColonyClient.md [hunk 3]
```diff
@@ -456,7 +456,7 @@ Submits a hidden work rating for a task. This is generated by generateSecret(_sa
 |---|---|---|
 |taskId|number|Integer taskId.|
 |role|Role|The role submitting their rating, either EVALUATOR or WORKER.|
-|ratingSecret|string|hidden work rating, generated as the output of `generateSecret(_salt, _rating)`, where `_rating` is a score from 0-50 (in increments of 10).|
+|secret|undefined|hidden work rating, generated as the output of `generateSecret(_salt, _rating)`, where `_rating` is a score from 1-3.|
```
Comment (JamesLefrere) [inline docs/_API_ColonyClient.md]: I guess this will all be fixed once it's in the docgen script.

File: docs/_API_ColonyClient.md (hunk 2 above)
Comment (chmanie) [reply docs/_API_ColonyClient.md]: Argl

**Pull request merged**